roles/sync_plans/README.md

```diff
@@ -0,0 +1,58 @@
+theforeman.foreman.sync_plans
+=============================
+
+This role defines Sync Plans.
+
+Role Variables
+--------------
+
+This role supports the [Common Role Variables](https://github.com/theforeman/foreman-ansible-modules/blob/develop/README.md#common-role-variables).
+
+- `sync_plans`: List of sync plans to create. Each sync plan is represented as a dictionary which specifies the `name` of the sync plan and the `products` assigned to the sync plan. It also specifies the `interval` which can be 'hourly', 'daily', 'weekly', or 'custom cron'. In case the 'custom cron' `interval` is used, it should also specify the `cron_expression`. Finally the sync plan should have a `sync_date` which specifies the first time that the sync plan will run.
+
+```yaml
+sync_plans:
+  - name: Weekly Sync
+    interval: weekly
+    sync_date: 2020-11-07 00:00:00 UTC
+    products:
+      - Red Hat Enterprise Linux Server
+      - Red Hat Software Collectiions (for RHEL Server)
+      - Red Hat Enterprise Linux for x86_64
+      - CentOS 8
+      - Debian 10
```

Comment:
how much work would it be to make this accept both, a list of strings, and a list of dicts? why you ask? then users can have one `products` definition and feed that to both `repositories` and `sync_plans` modules. at least for the simple case.

Reply:
(and your test playbook shows nicely how things get duplicated in simple cases)

Reply:
I decided against that approach for the reason that a product can be assigned to no sync plans, one sync plan, or multiple sync plans, and because the sync plan only needs to know the name of the assigned product and not the full details of its structure.

Specifying them together in a single structure only really makes sense imo in the case where there is a single sync plan with all products assigned. I think that is most likely to occur on a smaller deployment with a relatively fewer count of enabled products, and in that case, we really aren't saving that much work by preventing the user from listing the product names a second time under the sync plan.

Reply:
The other approach I considered was allowing the sync plan <-> product assignments to exist on the products data structure, but the benefits are again limited for the same reasons. Sync plans would still need to have their full attributes defined elsewhere, so we aren't really saving much work.

Ultimately I settled on the current approach because I think it is the easiest one to think about, and that to me is more important than a slight reduction in the amount of typing we are asking the user to do.

Reply:
Given there is really only that all products use case, I would lean towards wait for future enhancement if someone requests it to add that and leave this as a list. The notion that you have to supply the product name twice, and correctly twice is more a fault of the current design. And, separately, we could explore some documentation that shows how to define your data such that you can re-use names via yaml methods, e.g.

```
product_names:
  rhel_6: Red Hat Enterprise Linux 6
  rhel_7: Red Hat Enterprise Linux 7

products:
  - name: "{{ product_names['rhel_6'] }}"
  - name: "{{ product_names['rhel_7'] }}"
 
sync_plans:
   - name: Weekly Sync
     interval: weekly
     sync_date: 2020-11-07 00:00:00 UTC
     products:
       - "{{ product_names['rhel_6'] }}"
```

Reply:
Yeah, documentation is what I ended up in my head too. And it's even easier than what you have:

```
sync_plans:
   - name: Weekly Sync
     interval: weekly
     sync_date: 2020-11-07 00:00:00 UTC
     products: "{{ products | map(attribute='name') | list }}"
```

William, if you can incorporate that in the examples, I'm happy!

Reply:
Updated with additional documentation. Thanks for the suggestion on using yaml methods to avoid duplicating data.